tests/tracking/test_artifact_utils.py

```diff
@@ -45,3 +49,34 @@ def test_download_artifact_from_absolute_uri_persists_data_to_specified_output_d
     with open(os.path.join(
             artifact_output_path, logged_artifact_subdir, artifact_file_name), "r") as f:
         assert f.read() == artifact_text
+
+
+def test_upload_artifacts_to_databricks():
+    import_root = 'mlflow.tracking.artifact_utils'
+    with mock.patch(import_root + "._download_artifact_from_uri") as download_mock, \
+            mock.patch(import_root + ".DbfsRestArtifactRepository") as repo_mock:
+        new_source = _upload_artifacts_to_databricks('dbfs:/original/sourcedir/', 'runid12345',
+                                                     'databricks://tracking',
+                                                     'databricks://registry/ws')
+        download_mock.assert_called_once_with('dbfs://tracking@databricks/original/sourcedir/',
+                                              ANY)
+        repo_mock.assert_called_once_with(
+            'dbfs://registry:ws@databricks/databricks/mlflow/tmp-external-source/')
+        assert new_source == 'dbfs:/databricks/mlflow/tmp-external-source/runid12345/sourcedir'
```

Comment:
it'd be great to also check that `log_artifacts` is called with the right arguments, but i cannot figure out how to set the mock object / method correctly...